### DISCUSSION TITLE
Paired Analysis

### DISCUSSION THREAD

**Discussion by Nathanielhubert** [FrederickHuangLin/ANCOMBC#167] [Q&A] [answered]
I have 12 samples, collected from 6 sites before (pre) and after (post) a water rinse.
I want to know if there are differentially abundant differences between pre and post samples, but need to account for the paired samples not being independent.
How does the function know which is the pairing variable? 
By setting pairwise to true, does that define samples as being paired, or is that comparing multiple levels of factors in a pairwise manner?

I used the following command and the results look reasonable- I just want to confirm that I am performing and interpreting this correctly:

```
comp1_physeq_pseudocount_ancombc_paired <- ancombc2(data = comp1_physeq, 
                                                     fix_formula = "comp1_water_newPre_newPost_cat + comp1_water_newPre_newPost_pairs",
                                                     pseudo = 0.5,
                                                     group = "comp1_water_newPre_newPost_cat",
                                                     pairwise = TRUE)
```

Thank you

**Comment by FrederickHuangLin** [comment]
Hi @Nathanielhubert,

It seems your group variable of interest contains only two levels, pre and post. Given that, the `pairwise` argument is not necessary since it will only be activated when there are more than three groups. You can access the primary results in `res` output.

Best,
Huang

**Reply by Nathanielhubert** [reply]
Thank you, @FrederickHuangLin , for your answer.

I am a bit confused, however. I think after reviewing the pairwise element is comparing each treatment specified in a pairwise manner. And my goal is different:

I want to find which taxa are different between pre and post, but have pairs of samples, making them non-independent. I think that in order to avoid violating the assumption that samples are all independent of each other, I must somehow inform ANCOM-BC which samples were collected from the same site (are paired). I hoped that the test would thereby block by sampling site and tell me which taxa are different between pre and post after accounting for sampling site (pairs). But I now understand that is not the case. Do you know the appropriate test for doing this?

**Reply by FrederickHuangLin** [reply, marked as answer]
There may have been a miscommunication regarding the `pairwise` argument. Please note that the `pairwise` argument is intended for performing multi-group comparisons pairwisely, not for paired samples.

To elaborate, if you have 100 samples belonging to three groups (`A`, `B`, and `C`), setting `pairwise = TRUE` would enable `ancombc2` to identify differentially abundant taxa for pairwise comparisons such as `A-B`, `A-C`, and `B-C`.

In your case, the `pairwise` is irrelevant, as I mentioned earlier, since you only have two experimental groups. Since your samples are correlated with each other, instead, you can specify the `rand_formula` in `ancombc2` to perform a mixed-effects model. You can find more information about this in section 5 of the ANCOMBC2 (vignette)[https://bioconductor.org/packages/release/bioc/vignettes/ANCOMBC/inst/doc/ANCOMBC2.html].

Basically, I think you should specify a random intercept, (replace `subject_id` with your own id column)
```
rand_formula = "(1 | subject_id)"
```

Best,
Huang

**Reply by Nathanielhubert** [reply]
Thank you and I am sorry for the confusion!

**Reply by FrederickHuangLin** [reply]
Not a problem!
